salt/utils/network.py

```diff
@@ -199,31 +200,239 @@ def ip_to_host(ip):
+        'loopback': _ip_options_loopback,
+        'lo': _ip_options_loopback,
+        'multicast': _ip_options_multicast,
+        'private': _ip_options_private,
+        'public': _ip_options_global,
+        'reserved': _ip_options_reserved,
+        'site-local': _ip_options_site_local,
+        'sl': _ip_options_site_local,
+        'site_local': _ip_options_site_local,
+        'unspecified': _ip_options_unspecified
+    }
+
+    if not options:
+        return str(ip_obj)  # IP version already checked
+
+    options_list = [option.strip() for option in options.split(',')]
+
+    for option, fun in options_fun_map.items():
+        if option in options_list:
+            fun_res = fun(ip_obj, version)
+            if not fun_res:
+                return None
+                # stop at first failed test
+            # else continue
+    return str(ip_obj)
+
+
+def _is_ipv(ip, version, options=None):
+
+    if not version:
+        version = 4
+
+    if version not in (4, 6):
+        return None
+
     try:
-        return ipaddress.ip_address(ip).version == 4
+        ip_obj = ipaddress.ip_address(ip)
     except ValueError:
-        return False
+        # maybe it is an IP network
+        try:
+            ip_obj = ipaddress.ip_interface(ip)
+        except ValueError:
+            # nope, still not :(
+            return None
+
+    if not ip_obj.version == version:
+        return None
 
+    # has the right version, let's move on
+    return _ip_options(ip_obj, version, options=options)
 
-def is_ipv6(ip):
+
+def is_ipv4(ip, options=None):
     '''
-    Returns a bool telling if the value passed to it was a valid IPv6 address
+    Returns a bool telling if the value passed to it was a valid IPv4 address.
     '''
-    try:
-        return ipaddress.ip_address(ip).version == 6
-    except ValueError:
-        return False
+    return _is_ipv(ip, 4, options=options)
+
+
+def is_ipv6(ip, options=None):
+    '''
+    Returns a bool telling if the value passed to it was a valid IPv6 address.
+    '''
+    return _is_ipv(ip, 6, options=options)
+
+
+def _ipv_filter(value, version, options=None):
+    fun = None
+    if version == 4:
+        fun = is_ipv4
+    elif version == 6:
+        fun = is_ipv6
+
+    if not fun:  # indeed, that's not fun...
+        return
+
+    if isinstance(value, (six.string_types, six.text_type, six.binary_type)):
+        return fun(value, options=options)  # calls is_ipv4 or is_ipv6 for `value`
+    elif isinstance(value, (list, tuple, types.GeneratorType)):
+        # calls is_ipv4 or is_ipv6 for each element in the list
+        # os it filters and returns only those elements having the desired IP version
+        return [
+            fun(addr, options=options)
+            for addr in value
+            if fun(addr, options=options) is not None
+        ]
+    return None
+
+
+def ipv4(value, options=None):
+    '''
+    Filters a list and returns IPv4 values only.
```

Comment:
you really need another *ipv4* function for that ..? can't you just use a list comprehension ?